docs: Adding JS example of multiple foreign keys to the same table (#21827)
supabase/supabase: apps/docs/content/guides/database/joins-and-nesting.mdx

</TabPanel>
</Tabs>

## Specifying the `ON` clause for joins with multiple foreign keys

For example, if you have a project that tracks when employees check in and out of work shifts:

```sql
-- Employees
create table users (
  "id" serial primary key,
  "name" text
);

-- Badge scans
create table scans (
  "id" serial primary key,
  "user_id" int references users,
  "badge_scan_time" timestamp
);

-- Work shifts
create table shifts (
  "id" serial primary key,
  "user_id" int references users,
  "scan_id_start" int references scans, -- clocking in
  "scan_id_end" int references scans, -- clocking out
  "attendance_status" text
);
```

In this case, you need to explicitly define the join because the joining column on `shifts` is ambiguous.

> If we wanted to fetch all the `shifts` with `scan_id_start` related to a specific `scan`:

Reviewer: ```suggestion
To fetch all the `shifts` with `scan_id_start` related to a specific `scan`, use the following syntax:
```
